insights/specs/datasources/env.py

```diff
@@ -0,0 +1,129 @@
+    records the config file as an export file. When a line likes `unset LD_LIBRARY_PATH`
+    exists, the datasource records the config file as an unset file
+
+    The output of this datasource looks like:
+        export_files: /etc/environment, /root/.profile
+        unset_files: /etc/profile
+
+    Returns:
+        str: Returns at most 2 lines, one line contains the config files that define the
+             LD_LIBRARY_PATH, one line contains config files that unset the LD_LIBRARY_PATH.
+             Not return the line, because it might include sensitive information.
+
+    Raises:
+        SkipComponent: When any exception occurs.
+    """
+    env_name = "LD_LIBRARY_PATH"
+    config_files = [
+        "/etc/environment",
+        "/etc/env.d",
+        "/etc/profile",
+        "/etc/profile.d",
+        "/etc/bashrc",
+        "/etc/bash.bashrc",
+        "/root/.bash_profile",
+        "/root/.bashrc",
+        "/root/.profile",
+        "/root/.cshrc",
+        "/root/.zshrc",
+        "/root/.tcshrc"
+    ]
+    data = []
+
+    export_files = []
+    unset_files = []
+
+    for item in config_files:
+        if os.path.exists(item) and os.path.isfile(item):
+            if _is_env_exported(item, env_name):
+                export_files.append(item)
+            if _is_env_unset(item, env_name):
+                unset_files.append(item)
+        elif os.path.exists(item) and os.path.isdir(item):
+            for file_name in os.listdir(item):
+                # Only read the first level files here.
+                # Per the test, only the first level files will be executed.
+                file_path = os.path.join(item, file_name)
+                if os.path.isfile(file_path) and _is_env_exported(file_path, env_name):
+                    export_files.append(file_path)
+                if os.path.isfile(file_path) and _is_env_unset(file_path, env_name):
+                    unset_files.append(file_path)
+
+    if export_files:
+        data.append("export_files: " + ",".join(export_files))
+
+    if unset_files:
+        data.append("unset_files: " + ",".join(unset_files))
+
+    if not data:
+        raise SkipComponent()
+
+    return DatasourceProvider(content="\n".join(data), relative_path='insights_commands/ld_library_path_global_conf')
+
+
+def _is_env_exported(file_path, env_name):
+    # The logic here is:
+    # cat <config_file>|grep -w LD_LIBRARY_PATH|grep -w export
+    # To catch the following LD_LIBRARY_PATH env setting:
+    # 1. export LD_LIBRARY_PATH=/test/path
+    # 2. export     LD_LIBRARY_PATH
+    # 3. export LD_LIBRARY_PATH=/test/path TEST=/path/to/test
+    # There might be other syntaxs, they will not be collected.
+    pattern = r"\b{0}\b".format(env_name)
+
+    with open(file_path, 'r') as file:
+        for line in file.readlines():
+            line = line.strip()
+            if not line or line.startswith("#") or line.split()[0] != "export":
```

Comment:
To handle the comment lines, `line.startswith("#")` here is to skip the whole commented lines. The inline comment is another case that better be covered here, eg.   
```
export  TEST=/path/to/test   #  LD_LIBRARY_PATH=/test/path 
```
See the `Inline comment # comment`  handling in `get_active_lines()` .

Reply:
Thanks, updated.

Reply:
```
    with open(file_path, 'r') as file:
        active_lines = list(filter(None, (line.split("#", 1)[0].strip() for line in file.readlines())))
        for line in active_lines:
            if "export " in line and re.search(pattern, line):
                return True
```
@qinpingli Based on your newest implementation, what about calling this `get_active_lines()` func directly instead of rewrite the code?

Reply:
Hmm, I intend not calling `get_active_lines` to avoid potential circle import here, because looks like the `parsers` is a higher level package of `specs`,  it will import from `specs` package, so here I avoid to import `parsers` from `specs` package.